pkg/operator/starter.go

```diff
@@ -224,6 +225,13 @@ func prepareOauthOperator(controllerContext *controllercmd.ControllerContext, op
 
 	oauthInformers := oauthinformers.NewSharedInformerFactory(oauthClient, resync)
 
+	clusterVersionLister := operatorCtx.operatorConfigInformer.Config().V1().ClusterVersions().Lister()
+	clusterVersionConfig, err := clusterVersionLister.Get("version")
```

Comment:
I strongly recommend you add some debugging that prints out these two values (the clusterversion object and the err object), until you sort out what is going on (and then you can remove it)

Reply:
I bet my money that the issue is in the informer... 💸